tools/src/icon4py/tools/py2fgen/_definitions.py

```diff
@@ -94,7 +94,7 @@ class ScalarParamDescriptor(eve.Node):
     ArrayInfo: TypeAlias = tuple[Any, tuple[int, ...], bool, bool]
 
 if TYPE_CHECKING:
-    import cupy as cp  # type: ignore[import-untyped]
+    import cupy as cp  # type: ignore[import-not-found]
```

Comment:
The new type ignore kind of makes more sense than the previous one, I would explicitly explain why. IIUC the reason for this might be:
```suggestion
    import cupy as cp  # type: ignore[import-not-found]  # cupy might not be always available at type-checking 
```